lib/Util/CaseInsensitiveArray.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Stripe\Util;
+
+use ArrayAccess;
+
+class CaseInsensitiveArray implements ArrayAccess
```

Comment:
Do you think we could put a comment block in here that mentions that this exists to provide case insensitive access to response headers because their case may vary depending on whether the request came back over HTTP or HTTP/2 (which mandates lowercase)? I think it may be helpful in reasoning about why it's here in the future.